perllib/Open311/UpdatesBase.pm

```diff
@@ -158,20 +159,24 @@ sub process_update {
     my $external_status_code = $request->{external_status_code} || '';
     my $customer_reference = $request->{customer_reference} || '';
     my $old_external_status_code = $p->get_extra_metadata('external_status_code') || '';
-    my $comment = $self->schema->resultset('Comment')->new(
-        {
-            problem => $p,
-            user => $self->system_user,
-            external_id => $request->{update_id},
-            send_state => 'processed',
-            text => $self->comment_text_for_request(
-                $request, $p, $state, $old_state,
-                $external_status_code, $old_external_status_code
-            ),
-            confirmed => $request->{comment_time},
-            created => $request->{comment_time},
-        }
+    my ($text, $email_text) = $self->comment_text_for_request(
+                        $request, $p, $state, $old_state,
+                        $external_status_code, $old_external_status_code
+                        );
+    if (!$email_text && $request->{email_text}) {
+        $email_text = $request->{email_text} ? $request->{email_text} : '';
```

Comment:
Isn't `$request->{email_text}` always going to be set, based on the if statement?